src/ppom-storage.test.ts

```diff
@@ -0,0 +1,287 @@
+import { StorageBackend, PPOMStorage, StorageKey } from './ppom-storage';
+
+Object.defineProperty(globalThis, 'crypto', {
+  value: {
+    subtle: {
+      digest: () => new ArrayBuffer(12),
+    },
+  },
+});
+
+const checksum =
+  '409a7f83ac6b31dc8c77e3ec18038f209bd2f545e0f4177c2e2381aa4e067b49';
+class MockStorageBackend implements StorageBackend {
+  data: undefined;
+
+  constructor(data?: any) {
+    this.data = data;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async read(key: StorageKey): Promise<any> {
+    return new Promise((resolve) => {
+      resolve(this.data);
+    });
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async write(key: StorageKey, data: any): Promise<void> {
+    return new Promise((resolve) => {
+      resolve();
+    });
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async delete(key: StorageKey): Promise<void> {
+    return new Promise((resolve) => {
+      resolve();
+    });
+  }
+
+  async dir(): Promise<StorageKey[]> {
+    return new Promise((resolve) => {
+      resolve([]);
+    });
+  }
+}
+
+describe('PPOMStorage', () => {
+  it('should get initialised', () => {
+    const ppomStorage = new PPOMStorage({
+      storageBackend: new MockStorageBackend(),
+      readMetadata: () => [],
+      writeMetadata: () => undefined,
+    });
+    expect(ppomStorage).toBeDefined();
+  });
+
+  describe('PPOMStorage:readFile', () => {
```

Comment:
Nit: test names should already be prepended with the string passed to the outer `describe`, so you should be able to say:

``` suggestion
  describe('readFile', () => {
```